lib/commands/command-export-shell-version.bash

```diff
@@ -2,19 +2,50 @@
 # shellcheck source=lib/functions/versions.bash
 . "$(dirname "$(dirname "$0")")/lib/functions/versions.bash"
 
+unset_all_fish(){
+  printf "for var in (env | awk -F= '/^ASDF_[A-Za-z0-9_]+_VERSION=/ {print \$1}')\n"
+  printf "    echo Removing Env: \$var\n"
+  printf "    set -e \$var\n"
+  printf "end\n"
+}
+unset_all(){
+ printf "for var in \$(env | awk -F= '/^ASDF_[A-Za-z0-9_]+_VERSION=/ {print \$1}'); do\n"
+ printf "    echo Removing Env: " 
+ printf "    \$var\n"
+ printf "    unset \$var\n"
+ printf "done\n"
+}
+
 # Output from this command must be executable shell code
-shell_command() {
+shell_command() {     
   local asdf_shell="$1"
   shift
-
-  if [ "$#" -lt "2" ]; then
-    printf "Usage: asdf shell <name> {<version>|--unset}\n" >&2
+ 
+  if [ "$#" -lt "2" ] && [ "$1" != "--unset-all" ]; then
+    printf "Usage: asdf shell {<name> {<version>|--unset}|--unset-all}\n" >&2
     printf "false\n"
     exit 1
   fi
 
   local plugin=$1
   local version=$2
+  
+  if [ "$plugin" = "--unset-all" ] ; then
```

Comment:
to the top of this if? 
![image](https://github.com/asdf-vm/asdf/assets/75655486/fa41162f-5bb6-417a-b0d9-2736b480396c)

Reply:
Sorry, I meant directly under the help if-block. The logic for printing the help menu should come first if possible.

_Please_ make sure you install [EditorConfig](https://editorconfig.org) for you IDE and re-save the file. I noticed there are some extra trailing whitespace characters.

Reply:
I normally use neovim, it comes with native EditorConfig support, but I tried to update some things in vscodium and I was testing it I think the indetation failed because of that

Reply:
and sorry I still don't understand where I should put the if

Reply:
It's fine, I guess it's kind of a nit anyways

Reply:
about the if, where i would put it?

Reply:
Yeah it can be closed. I only slightly preferred it to be at the top so #1456 would be a bit more consistent

Reply:
I did it, check it now, pls